### PR TITLE
8235834: IBM-943 charset encoder needs updating

### PR DESCRIPTION
**Pull request by psawant19**
Backporting JDK-8235834 - IBM-943 charset encoder needs updating.
There are few characters which display garbled values when using IBM943C, so this mapping file needs to be backported to fix the issue.

Reported issue : https://bugs.openjdk.org/browse/JDK-8235834

---------
### Progress
- [x] Change must be properly reviewed (1 review required, with at least 1 [Reviewer](https://openjdk.org/bylaws#reviewer))
- [x] [JDK-8235834](https://bugs.openjdk.org/browse/JDK-8235834) needs maintainer approval
- [x] Change must not contain extraneous whitespace
- [x] Commit message must refer to an issue

### Issue
 * [JDK-8235834](https://bugs.openjdk.org/browse/JDK-8235834): IBM-943 charset encoder needs updating (**Bug** - P3 - Approved)

### Reviewers
 * [Severin Gehwolf](https://openjdk.org/census#sgehwolf) (@jerboaa - **Reviewer**)

### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk8u-dev.git pull/476/head:pull/476` \
`$ git checkout pull/476`

Update a local copy of the PR: \
`$ git checkout pull/476` \
`$ git pull https://git.openjdk.org/jdk8u-dev.git pull/476/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 476`

View PR using the GUI difftool: \
`$ git pr show -t 476`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk8u-dev/pull/476.diff">https://git.openjdk.org/jdk8u-dev/pull/476.diff</a>

</details>

### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk8u-dev/pull/476#issuecomment-2081881704)

### CODE REVIEW

**Comment by psawant19** [comment]
This contribution is on behalf of my employer, IBM, which is a corporate OCA signatory.

**Comment by psawant19** [comment]
Closing this PR as my OCA verification is taking too long to approve. 
Opened another PR by a different contributor:- https://github.com/openjdk/jdk8u-dev/pull/480.

**Comment by psawant19** [comment]
Backporting JDK-8235834 - IBM-943 charset encoder needs updating.
There are few characters which display garbled values when using IBM943C, so this mapping file needs to be backported to fix the issue.

Reported issue : https://bugs.openjdk.org/browse/JDK-8235834.

**Comment by psawant19** [comment]
@gnu-andrew, @mrserb Could you please review this PR.

**Comment by offamitkumar** [comment]
Hi @psawant19 , 
This PR doesn't make any sense if https://github.com/openjdk/jdk8u-dev/pull/480 is in review. Please close one of them.

File: jdk/test/sun/nio/cs/TestIBMBugs.java
Comment (offamitkumar) [inline jdk/test/sun/nio/cs/TestIBMBugs.java:27]: Please make sure that you're only backporting the "changes" done by the head-stream patch. Not more than that (if not required). See my comments on #480 .

**Comment by psawant19** [comment]
/clean The patch is identical to the patch in jdk11u-dev.

**Comment by offamitkumar** [comment]
Please merge master branch and that should trigger the GHA test again.

**Comment by offamitkumar** [comment]
>/integrate

@psawant19, Please go through the contribution guidelines. Don't just integrate!!

There is progress check list on your PR. Once all of them are marked as `completed`. Then only you should go for integration. 

So for now, You need: 
1.  approval from **R**ivewer. 
2. approval from Maintainer.

>Failures are a Certigna certificate test and an Ubuntu packages not available at specified path. Hence integrating.

I asked you to merge master branch in `psawant19:IBM943-support-backport-1b9251` so that Github will run the test cases again. Please do it!!

**Comment by offamitkumar** [comment]
I guess something went wrong with GHA test. Have you turned off Github Actions ? 

@RealCLanger is it possible to get review from you ?

**Review by jerboaa — CHANGES_REQUESTED** [review]
The test fails:

```
/disk/openjdk/upstream-sources/git/jdk8u-dev/jdk/test/sun/nio/cs/TestIBMBugs.java:223: error: cannot find symbol
            if (!Arrays.equals(byteIBM943c2b, strIBM943c2b1.getBytes(cs))) {
                 ^
  symbol:   variable Arrays
  location: class TestIBMBugs
1 error
result: Failed. Compilation failed: Compilation failed
```

Adding the import should fix it.

Also, please enable GHA for your fork and show some evidence on what testing you've done for this. Thanks!

**Comment by psawant19** [comment]
@jerboaa,
I have fixed the compilation error and GHAs are enabled on this repo.
IBM943 mapping document link: https://www.fileformat.info/info/charset/x-IBM943/list.htm

Attaching test evidences for the bug fix with before and after screenshots.

Before fix:
image

After fix:
image

**Comment by jerboaa** [comment]
@psawant19 Thanks. Please also show that the `sun/nio/cs` jtreg tests are passing.

**Review by jerboaa — APPROVED** [review]
This looks OK.

**Comment by jerboaa** [comment]
> I have fixed the compilation error and GHAs are enabled on this repo.

Please merge lastest master into your branch, which should make the certinga CA tests failure go away.

**Comment by psawant19** [comment]
@jerboaa,
There is still 1 pre-test failing that is because these below packages doesn't exist at this location. 
- https://httpredir.debian.org/debian/dists/buster/main/binary-s390x/Packages
- https://httpredir.debian.org/debian/dists/buster/main/binary-ppc64el/Packages

Is the PR good to go for maintainer approval?

**Comment by offamitkumar** [comment]
>Is the PR good to go for maintainer approval?


Yes, use `/approval request <your reason>` to apply for the approval. Please give justification why you want to backport the changes, what testing you have done and what is risk (low/high). In this case it should be low.

**Comment by psawant19** [comment]
/approval request This back port is required to display Japanese locale which uses IBM943 charset. The fix is tested in a linux environment and the test case passes as mentioned in the screenshot above.

**Comment by jerboaa** [comment]
The follow-up, [JDK-8239965](https://bugs.openjdk.org/browse/JDK-8239965), needs backporting too. Please do that so that I can approve them as a unit.

**Comment by psawant19** [comment]
This is the PR (https://github.com/openjdk/jdk8u-dev/pull/500) created for backporting [JDK-8239965](https://bugs.openjdk.org/browse/JDK-8239965).

**Pull request closed without merging**
